ETHDINFKBot/Handlers/MessageHandler.cs

```diff
@@ -297,7 +299,14 @@ private async void EmojiDetection()
 
                                 using (var stream = new MemoryStream(File.ReadAllBytes(emote.LocalPath)))
                                 {
-                                    await SocketMessage.Channel.SendFileAsync(stream, $"{emote.EmoteName}.gif");
+                                    if (Context.Message.ReferencedMessage != null)
+                                    {
+                                        await Context.Channel.SendFileAsync(stream, $"{emote.EmoteName}.gif", "", false, null, null, false, null, new Discord.MessageReference(Context.Message.ReferencedMessage.Id));
+                                    } 
+                                    else
+                                    {
+                                        await SocketMessage.Channel.SendFileAsync(stream, $"{emote.EmoteName}.gif");
```

Comment:
As Discord.MessageReference can accept null you could do

`await SocketMessage.Channel.SendFileAsync(stream, $"{emote.EmoteName}.gif", "", false, null, null, false, null, new Discord.MessageReference(SocketMessage.ReferencedMessage?.Id));`

and remove the if statement to simplify code